application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -407,18 +403,20 @@ record TemporaryData(Instant expiresAt, String duration) {
      * @param temporaryData if the action is a temporary action, such as a 1 hour mute.
      * @param guild for which the action was triggered.
      * @param reason for the action.
+     * @param textChannel for which messages are being sent to.
+     *
      * @return the appropriate advice.
      */
-    public static RestAction<MessageCreateAction> sendDmAdvice(ModerationAction action,
-            @Nullable ModerationUtils.TemporaryData temporaryData, Guild guild, String reason,
+    public static RestAction<?> sendDmAdvice(ModerationAction action,
```

Comment:
dont return `?`, otherwise the call-site cant use it type-safe at all.

return what `sendMessage` actually returns, namely `MessageAction` instead of `RestAction<?>`. (which is a `RestAction<Message>`)

Reply:
@DevSerendipity `RestAction<Message>` is okay, but `MessageAction` would be even better. can u change the type to that please? :)

Reply:
I don't have `MessageAction `